src/core/Tree/Tree.tsx

```diff
@@ -0,0 +1,215 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { CommonProps, useTheme } from '../utils';
+import '@itwin/itwinui-css/css/tree.css';
+import cx from 'classnames';
+
+export const TreeContext = React.createContext<
+  | {
+      nodeDepth: number;
+      subNodeIds?: string[];
+      hasSubNodes: boolean;
+    }
+  | undefined
+>(undefined);
+
+export type NodeData<T> = {
```

Comment:
Add comments to each property so users would know what each of them means.

Reply:
Added comments